src/search/utils/strings.cc

```diff
@@ -8,6 +8,44 @@
 using namespace std;
 
 namespace utils {
+string escape(const string &s) {
+    string result;
+    result.reserve(s.length());
+    for (char c : s) {
+        if (c == '\\') {
+            result += "\\\\";
+        } else if (c == '"') {
+            result += "\\\"";
+        } else if (c == '\n') {
+            result += "\\n";
+        } else {
+            result += c;
+        }
+    }
+    return result;
+}
+
+string unescape(const string &s) {
```

Comment:
Is it worth mentioning in a code comment that this will not do syntax-checking and therefore accepts "illegal" strings? Illegal strings are those where the final character is an unescaped "\" or where an escaping "\" is not followed by one of the characters from the specific set that can follow an escaping "\".